### PR TITLE
WIP UserService

### PR DESCRIPTION
**Pull request by PavelBoPavel**

### CODE REVIEW

**Review by trump-ny — APPROVED** [review]

**Review by VasiliZas — APPROVED** [review]

File: homework/PavelBo/loops/userLoginService/src/main/java/com/tutrit/java/quickstart/User.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package com.tutrit.java.quickstart;
+
+import java.util.Objects;
+
+public class User {
+
+    private String login;
+    private String password;
+    private boolean BLOCKED;
```
Comment (KaydunovDenis) [inline homework/PavelBo/loops/userLoginService/src/main/java/com/tutrit/java/quickstart/User.java:9]: https://www.oracle.com/technetwork/java/codeconventions-150003.pdf

BLOCKED -> isBlocked
переменная пишется заглавными буквами когда она
The names of variables declared class con-stants  and of ANSI constants should be alluppercase with words separated by under-scores (“_”). (ANSI constants should beavoided, for ease of debugging.)
-> public static final

File: homework/PavelBo/loops/userLoginService/src/main/java/com/tutrit/java/quickstart/User.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        if (this == o) return true;
+        if (!(o instanceof User)) return false;
+        User user = (User) o;
+        return Objects.equals(getLogin(), user.getLogin()) && Objects.equals(getPassword(), user.getPassword());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getLogin(), getPassword());
+    }
```
Comment (KaydunovDenis) [inline homework/PavelBo/loops/userLoginService/src/main/java/com/tutrit/java/quickstart/User.java:53]: move duwn this methods and add toString

File: homework/PavelBo/loops/userLoginService/src/main/java/com/tutrit/java/quickstart/User.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    public void resetAttempts () {
+        аttempts = 3;
+    }
+}
```
Comment (KaydunovDenis) [inline homework/PavelBo/loops/userLoginService/src/main/java/com/tutrit/java/quickstart/User.java:58]: move up this methods

File: homework/PavelBo/loops/userLoginService/src/main/java/com/tutrit/java/quickstart/UserLoginService.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package com.tutrit.java.quickstart;
+
+public class UserLoginService {
+
+    User user;
```
Comment (KaydunovDenis) [inline homework/PavelBo/loops/userLoginService/src/main/java/com/tutrit/java/quickstart/UserLoginService.java:5]: make private

**Review by KaydunovDenis — CHANGES_REQUESTED** [review]

File: homework/PavelBo/loops/userLoginService/src/test/java/com/tutrit/java/quickstart/UserLoginServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        Assert.assertEquals(false, user.isBLOCKED());
+        userLoginService.login(user, "1235");
+        Assert.assertEquals(false, user.isBLOCKED());
+        userLoginService.login(user, "1235");
+        Assert.assertEquals(true, user.isBLOCKED());
+        userLoginService.login(user, "1234");
+        Assert.assertEquals(true, user.isBLOCKED());
+    }
+
+}
```
Comment (max-java) [inline homework/PavelBo/loops/userLoginService/src/test/java/com/tutrit/java/quickstart/UserLoginServiceTest.java:71]: please, add empty row

File: homework/PavelBo/loops/userLoginService/src/main/java/com/tutrit/java/quickstart/UserLoginService.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    public boolean login(User user, String password) {
+        if (user.isBLOCKED()){
+            return false;
+        }else {
+            if (isPassword(user, password)){
+                user.resetAttempts();
+                return true;
+            }else {
+                user.setАttempts(user.getАttempts() - 1);
+                blockedUser(user);
```
Comment (max-java) [inline homework/PavelBo/loops/userLoginService/src/main/java/com/tutrit/java/quickstart/UserLoginService.java:20]: `blockedUser(user);` -> `blockUser(user);`will be more readable

also, I suggest that you use Atl+Ctrl +L before merge

File: homework/PavelBo/loops/userLoginService/src/main/java/com/tutrit/java/quickstart/UserLoginService.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        }
+    }
+
+    private void blockedUser(User user) {
+        if (user.getАttempts() == 0){
+            user.setBLOCKED(true);
+        }
+    }
+
+    private boolean isPassword(User user, String password) {
```
Comment (max-java) [inline homework/PavelBo/loops/userLoginService/src/main/java/com/tutrit/java/quickstart/UserLoginService.java:32]: `isPassword` -> `verifyPassword` more readable

**Review by max-java — CHANGES_REQUESTED** [review]
ptal, and see colleagues comments.

**Pull request closed without merging**